chromium_src/components/search_engines/template_url_prepopulate_data.cc

```diff
@@ -95,19 +83,40 @@ const BravePrepopulatedEngineID brave_engines_FR[] = {
     PREPOPULATED_ENGINE_ID_STARTPAGE,
 };
 
+// Australia / NZ / IE - DDG appears on top.
+std::vector<BravePrepopulatedEngineID> brave_engines_AU_NZ_IE = {
+    PREPOPULATED_ENGINE_ID_DUCKDUCKGO_AU_NZ_IE,
+#if defined(OS_ANDROID)
+    PREPOPULATED_ENGINE_ID_DUCKDUCKGO_LITE,
+#endif
+    PREPOPULATED_ENGINE_ID_GOOGLE,
+    PREPOPULATED_ENGINE_ID_QWANT,
+    PREPOPULATED_ENGINE_ID_BING,
+    PREPOPULATED_ENGINE_ID_STARTPAGE};
+
+// A map to keep track of default engines for countries that don't use the
+// regular default engine.
+const std::map<int, std::vector<BravePrepopulatedEngineID>>
+    default_engines_by_country_id_map = {
+        {country_codes::CountryCharsToCountryID('D', 'E'), brave_engines_DE},
+        {country_codes::CountryCharsToCountryID('F', 'R'), brave_engines_FR},
+        {country_codes::CountryCharsToCountryID('A', 'U'),
+         brave_engines_AU_NZ_IE},
+        {country_codes::CountryCharsToCountryID('N', 'Z'),
+         brave_engines_AU_NZ_IE},
+        {country_codes::CountryCharsToCountryID('I', 'E'),
+         brave_engines_AU_NZ_IE}};
+
 // Builds a vector of PrepulatedEngine objects from the given array of
 // |engine_ids|. Fills in the default engine index for the given |country_id|,
 // if asked.
 std::vector<const PrepopulatedEngine*> GetEnginesFromEngineIDs(
-    const BravePrepopulatedEngineID engine_ids[],
-    size_t num_ids,
+    const std::vector<BravePrepopulatedEngineID> engine_ids,
```

Comment:
const std::vector<BravePrepopulatedEngineID>`&` engine_ids ? (will need to use enginge_ids.at(i) instead of [i] later in the code).

Reply:
Fixed with https://github.com/brave/brave-core/pull/3591/commits/201436c15a1660d11fc108ac1495933e2f8358ea 👍